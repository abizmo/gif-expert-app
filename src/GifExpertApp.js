import React, {useState} from 'react'
import PropTypes from 'prop-types'

import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultState }) => {
  const [categories, setCategories] = useState(defaultState)

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory addCategory={setCategories} />
      <hr />
      <ol>
        { 
          categories.map(category => (
            <GifGrid key={category} category={category} />
          ))
        }
      </ol>
    </div>
  )
}

GifExpertApp.propTypes = {
  defaultState: PropTypes.arrayOf(PropTypes.string),
}

GifExpertApp.defaultProps = {
  defaultState: []
}

export default GifExpertApp
