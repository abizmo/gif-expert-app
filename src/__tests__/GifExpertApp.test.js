import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('GifExpertApp Tests', () => {
  test('should render and match with the snapshot', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should show a list of categories', () => {
    const categories = ['fail', 'paint']
    const wrapper = shallow(<GifExpertApp defaultState={categories} />)

    expect(wrapper).toMatchSnapshot()
    expect( wrapper.find('GifGrid')).toHaveLength(categories.length)
  })
  
})
