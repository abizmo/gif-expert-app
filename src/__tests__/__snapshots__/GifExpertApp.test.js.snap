// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Tests should render and match with the snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`GifExpertApp Tests should show a list of categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    addCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="fail"
      key="fail"
    />
    <GifGrid
      category="paint"
      key="paint"
    />
  </ol>
</div>
`;
